### PR TITLE
Re-sharing

### PR DESCRIPTION
**Pull request by pavelkrolevets**
- Add reshare with eip1271 signature verification according to spec
- Add threshold of old operators support
- Use types from dkg-spec and align
- Add integration tests for resharing cases
- Add docker demo for resharing
- Update re-signing to spec (eip1271 sig)
- Add more integration resign tests

### CODE REVIEW

File: cli/initiator/reshare.go
Comment (radiken) [inline cli/initiator/reshare.go:150]: Should be resharing

File: cli/initiator/resigning.go
Comment (radiken) [inline cli/initiator/resigning.go:103]: Should be resignMsg

File: pkgs/initiator/initiator.go
Comment (radiken) [inline pkgs/initiator/initiator.go:514]: Can wrap these in another function as it is the same for init, resign, reshare (also commented in pull request 100)

File: pkgs/dkg/drand.go
Comment (radiken) [inline pkgs/dkg/drand.go:709]: Why do we use spec.Result as the output data for Resign function above (line 639-655) but not here (and init)?

**Review by alan-ssvlabs — CHANGES_REQUESTED** [review]
We have a security report that requires discussion before approval.

File: pkgs/dkg/drand.go
Comment (pavelkrolevets) [reply pkgs/dkg/drand.go:650]: At reshare we have more steps, but at resign only one

File: pkgs/dkg/drand.go
Comment (pavelkrolevets) [reply pkgs/dkg/drand.go:410]: we need an empty object encoded

File: pkgs/dkg/drand.go
Comment (pavelkrolevets) [reply pkgs/dkg/drand.go:782]: need to refactor this function, ill try, I dont like how it is atm

File: pkgs/initiator/initiator.go
Comment (pavelkrolevets) [reply pkgs/initiator/initiator.go:242]: yeah, good catch, some leftovers after hundreds of iterations

File: pkgs/operator/state.go
Comment (pavelkrolevets) [reply pkgs/operator/state.go]: done at resigning, merged here

File: pkgs/operator/state.go
Comment (pavelkrolevets) [reply pkgs/operator/state.go:158]: I refactored here, so we have now
```
	case *wire.ReshareMessage:
		commits, share, err := s.getPublicCommitsAndSecretShare(msg)
		owner.SecretShare = share
		resp, err = owner.Reshare(reqID, &msg.SignedReshare.Reshare, commits)
		if err != nil {
			return nil, nil, err
		}
```

Comment (pavelkrolevets) [reply pkgs/operator/state.go:158]: probalby cant move this before `owner` creation

File: pkgs/initiator/initiator.go
Comment (pavelkrolevets) [reply pkgs/initiator/initiator.go:183]: do you want to move it inside? I thought it would be more readable that we check threshold each round of replies

**Pull request merged**
